src/tile/models/tileManager.ts

```diff
@@ -0,0 +1,166 @@
+      }
+
+      const params = this.getRenderParams(zoom, lon, lat);
+
+      renderer.render(params, (error, buffer) => {
+        this.renderersPool.release(renderer);
+
+        if (error) {
+          return errorCallback(error);
+        }
+
+        this.removeSemiTransparentOutlinesOnBuffer(buffer);
+
+        sharp(buffer, {
+          raw: {
+            width: params.width as number,
+            height: params.height as number,
+            channels: NUM_OF_CHANNELS_IN_BUFFER,
+          },
+        })
+          .png({ adaptiveFiltering: false })
+          .toBuffer({ resolveWithObject: false })
+          .then(successCallback)
+          .catch(errorCallback);
+      });
+    });
+  }
+
+  private getRenderParams(zoom: number, lon: number, lat: number): RenderParams {
+    const { tileSize, zoom: zoomSettings } = this.application;
+    return {
+      zoom: Math.max(zoomSettings.min, tileSize === DEFAULT_TILE_SIZE ? zoom - 1 : zoom),
+      center: [lon, lat],
+      width: tileSize,
+      height: tileSize,
+    };
+  }
+
+  private removeSemiTransparentOutlinesOnBuffer(buffer: Buffer): void {
+    for (let i = 0; i < buffer.length; i += NUM_OF_CHANNELS_IN_BUFFER) {
+      const alpha = buffer[i + BufferChannel.ALPHA];
+      if (alpha === MAX_ALPHA_VALUE) {
+        continue;
+      }
+      const norm = alpha / MAX_ALPHA_VALUE;
+
+      // if fully-transparent change to black, otherwise normalize the color
+      if (alpha === 0) {
+        buffer[i + BufferChannel.RED] = BLACK;
+        buffer[i + BufferChannel.GREEN] = BLACK;
+        buffer[i + BufferChannel.BLUE] = BLACK;
+      } else {
+        buffer[i + BufferChannel.RED] /= norm;
+        buffer[i + BufferChannel.GREEN] /= norm;
+        buffer[i + BufferChannel.BLUE] /= norm;
+      }
+    }
+  }
+
+  private createPool(ratio: number, minResources: number, maxResources: number): Pool<Map> {
+    const createRenderer = (ratio: number, createCallback: (err: Error | null, renderer: Map) => void): void => {
+      const renderer = new Map({ ...this.renderHandler.renderOptions, ratio });
+      renderer.load(this.global.styleContent);
+      createCallback(null, renderer);
+    };
+    return new Pool({
+      min: minResources,
+      max: maxResources,
+      create: createRenderer.bind(null, ratio),
+      destroy: (renderer: Map): void => {
+        renderer.release();
+      },
+    });
+  }
+
+  private validateTileBounds(z: number, x: number, y: number): void {
+    if (!this.isZoomValid(z) || !this.isAxisTileValid(x, z) || !this.isAxisTileValid(y, z)) {
+      throw new OutOfBoundsError(`tile request for z: ${z}, x: ${x}, y: ${y} is out of bounds.`);
+    }
+  }
+
+  private isZoomValid(z: number): boolean {
+    const { zoom } = this.application;
+    return z >= zoom.min && z <= zoom.max;
+  }
+
+  private isAxisTileValid(axisTileNumber: number, z: number): boolean {
+    return axisTileNumber >= 0 && axisTileNumber <= POWERS_OF_TWO_PER_ZOOM_LEVEL[z];
+  }
```

Comment:
the functions do not need any property from `this` -> function can be exported to the module level, 
then `this.validateTileBounds(z, x, y);` -> 
```ts
if (!isTileInBound) {
     throw new OutOfBoundsError(`tile request for z: ${z}, x: ${x}, y: ${y} is out of bounds.`);
}
```

Reply:
done.